app/src/helpers/lockdrop/EthereumLockdrop.ts

```diff
@@ -117,33 +92,175 @@ export async function getCurrentAccountLocks(
     }
 }
 
-// returns an array of the entire list of locked events for the contract only once
-export async function getAllLockEvents(web3: Web3, instance: Contract): Promise<LockEvent[]> {
-    // this value can be set as the block number of where the contract was deployed
-    const startBlock = 0;
+export function defaultAffiliation(aff: string) {
+    // check if affiliation address is not empty and is not themselves
+    if (aff) {
+        // return itself when it is a valid address
+        return aff;
+    } else {
+        // if it is an invalid address, return the default affiliation address
+        return defaultAddress;
+    }
+}
+
+export async function getCurrentUsdRate() {
+    let usdRate = 0;
     try {
-        // get all the event data
-        const ev = await instance.getPastEvents('Locked', { fromBlock: startBlock });
+        const res = await fetch(ethMarketApi);
+        const data = await res.json();
+        usdRate = data.market_data.current_price.usd;
+    } catch (error) {
+        console.log(error);
+    }
+    return usdRate;
+}
 
-        return Promise.all(
-            ev.map(async i => {
-                const transactionString = await Promise.resolve(web3.eth.getBlock(i.blockNumber));
-                const time = transactionString.timestamp.toString();
+async function plmBaseIssueAmount(lockData: LockEvent) {
```

Comment:
I think it has missed. Let's pair-programming and double-check tomorrow with me.
![image](https://user-images.githubusercontent.com/6259384/79069468-79364980-7d09-11ea-80f5-a3e43bbf10c4.png)